fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java

```diff
@@ -382,6 +382,33 @@ public static List<ScalarOperator> getAllPredicates(OptExpression root) {
         return predicates;
     }
 
+    // If join is not cross/inner join, MV Rewrite must rewrite, otherwise may cause bad results.
+    // eg.
+    // mv: select * from customer left outer join orders on c_custkey = o_custkey;
+    //
+    //query（cannot rewrite）:
+    //select count(1) from customer left outer join orders
+    //  on c_custkey = o_custkey and o_comment not like '%special%requests%';
```

Comment:
why would user write such SQL

Reply:
Yep.... We cannot decide it. But for non-inner join, we should take care of on-predicates or where-predicates.